packages/block-library/src/image/index.php

```diff
@@ -14,6 +14,15 @@
  * @return string Returns the block content with the data-id attribute added.
  */
 function render_block_core_image( $attributes, $content ) {
+
+	// Don't render if there is no URL set.
+	$processor = new WP_HTML_Tag_Processor( $content );
+	$processor->next_tag( 'img' );
+
+	if ( $processor->get_attribute( 'src' ) === null ) {
+		return '';
+	}
```

Comment:
Is there are the reason we're not checking if `$attributes['url']` is empty or not?

Reply:
My understanding is that the `src` attribute is not included in the `$attirbutes` because the `selector` is defined in the `src` attribute.

Within the render callback function, `$attributes` variable is like this:

```
array(4) {
  ["id"]=>
  int(21)
  ["sizeSlug"]=>
  string(4) "full"
  ["linkDestination"]=>
  string(4) "none"
  ["alt"]=>
  string(0) ""
}
```

So I thought we would need to parse `$contents` to determine if there is an image or not.

Reply:
Good point attributes derived from HTML sources aren’t available in array.